### PR TITLE
Python rtsp

### PR DESCRIPTION
**Pull request by brian-intel**
## PR Checklist

- [x] Added label to the Pull Request for easier discoverability and search
- [x] Commit Message meets guidelines as indicated in the URL https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md
- [x] Every commit is a single defect fix and does not mix feature addition or changes
- [x] Unit Tests have been added for new changes
- [ ] Updated Documentation as relevant to the changes
- [ ] All commented code has been removed
- [ ] If you've added a dependency, you've ensured license is compatible with repository license and clearly outlined the added dependency.
- [ ] PR change contains code related to security
- [ ] PR introduces changes that breaks compatibility with other modules (If YES, please provide details below)

## What are you changing?

feat: Adding a more generic python OVMS pipeline

## Issue this PR will close

close: #**issue_number**

## Test Instructions if applicable
- run camera-simulator
- run server using: sudo ./docker-run.sh --workload opencv-ovms --platform core --inputsrc rtsp://127.0.0.1:8554/camera_0
- stop the client container: 'docker rm ovms-client0'
- run container:
docker run --network host --privileged --user root --privileged --ipc=host --name ovms-client0 -e RENDER_MODE= -e INPUTSRC_TYPE=RTSP -e DISPLAY=:1 -e cl_cache_dir=/home/pipeline-server/.cl-cache -v /home/intel/projects/automated-self-checkout/.cl-cache:/home/pipeline-server/.cl-cache -v /tmp/.X11-unix:/tmp/.X11-unix -v /home/intel/projects/automated-self-checkout/sample-media/:/home/pipeline-server/vids -v /home/intel/projects/automated-self-checkout/configs/pipelines:/home/pipeline-server/pipelines -v /home/intel/projects/automated-self-checkout/configs/extensions:/home/pipeline-server/extensions -v /home/intel/projects/automated-self-checkout/results:/tmp/results -v /home/intel/projects/automated-self-checkout/configs/opencv-ovms/images:/images -v /home/intel/projects/automated-self-checkout/configs/opencv-ovms/scripts:/scripts -v /home/intel/projects/automated-self-checkout/configs/opencv-ovms/models/2022:/models -v /home/intel/projects/automated-self-checkout/configs/framework-pipelines:/home/pipeline-server/framework-pipelines -it -e GRPC_PORT=9000  ovms-client:latest bash
- run 'python3 /scripts/grpc_python_test.py'

### CODE REVIEW

File: configs/opencv-ovms/scripts/grpc_postprocess.py [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+    output3 = as_numpy(response, "masks")
+    nu = np.array(output1)
+    nu2 = np.array(output2)
+    nu3 = np.array(output3)
+    # for object classification models show imagenet class
+    print('Processing time: {:.2f} ms; fps: {:.2f}'.format(round(np.average(duration), 2),round(1000  / np.average(duration), 2)))
+    return output1
+
+def postProcessBit(response, duration):
+    # omz instance segmentation model has three outputs
```
Comment (jim-wang-intel) [inline configs/opencv-ovms/scripts/grpc_postprocess.py:55]: is this comment accurate?

File: configs/opencv-ovms/scripts/grpc_python.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+import grpc
+from client_utils import print_statistics
+from tritonclient.grpc import service_pb2, service_pb2_grpc
+from grpc_postprocess import *
+
+def openInputSrc(input_src):
+    # OpenCV RTSP Stream
+    stream = cv2.VideoCapture(input_src)
+    if not stream.isOpened():
+        print('Cannot open RTSP stream')
```
Comment (jim-wang-intel) [inline configs/opencv-ovms/scripts/grpc_python.py:21]: add debugging logs about what `input_src` could not be opened

File: configs/opencv-ovms/scripts/grpc_python_test.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        _, frame = stream.read()
+        img = cv2.resize(frame, (model_size[0], model_size[1]))
+        img_str = cv2.imencode('.jpg', img)[1].tobytes()
+        # Get GRPc
+        grpc_stub = setupGRPC("127.0.0.1", "9000")
+        response = inference(img_str,"instance-segmentation-security-1040", grpc_stub)
+        results = postProcessMaskRCNN(response[0], response[1])
+        self.assertTrue(len(results))
+
+# class TestPostProcessBit(unittest.TestCase):
```
Comment (jim-wang-intel) [inline configs/opencv-ovms/scripts/grpc_python_test.py:69]: Add a TODO comment for this one

**Review by jim-wang-intel — CHANGES_REQUESTED** [review]

File: configs/opencv-ovms/scripts/grpc_postprocess.py (hunk 1 above)
Comment (brian-intel) [reply configs/opencv-ovms/scripts/grpc_postprocess.py:55]: Will remove

File: configs/opencv-ovms/scripts/grpc_python.py (hunk 2 above)
Comment (brian-intel) [reply configs/opencv-ovms/scripts/grpc_python.py:21]: Adding more log data

File: configs/opencv-ovms/scripts/grpc_python_test.py (hunk 3 above)
Comment (brian-intel) [reply configs/opencv-ovms/scripts/grpc_python_test.py:69]: done

File: configs/opencv-ovms/scripts/grpc_python.py [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    return stream
+
+def setupGRPC(address, port):
+    address = "{}:{}".format(address, port)
+    # Create gRPC stub for communicating with the server
+    channel = grpc.insecure_channel(address)
+    grpc_stub = service_pb2_grpc.GRPCInferenceServiceStub(channel)
+    return grpc_stub
+
+def getModelSize(model_name):
```
Comment (marcpfuller) [inline configs/opencv-ovms/scripts/grpc_python.py:33]: is this temporary? I thought the plan was to get the model size from the ovms config file....

File: configs/opencv-ovms/scripts/grpc_python.py [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    stream = openInputSrc(args['input_src'])
+
+    print("Establish OVMS GRPc connection")
+    grpc_stub = setupGRPC(args['grpc_address'],args['grpc_port'])
+
+    print("Get the model size from OVMS metadata")
+    model_size = getModelSize(args['model_name'])
+
+    print("Begin inference loop")
+    while True:
```
Comment (ejlee3) [inline configs/opencv-ovms/scripts/grpc_python.py:75]: use a try-except block here so that any errors can be caught without breaking the loop.

**Review by jim-wang-intel — COMMENTED** [review]
LGTM except fixing the review dog's issue

File: configs/opencv-ovms/scripts/grpc_python.py [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+        try:
+            # get frame from OpenCV
+            _, frame = stream.read()
+            img = cv2.resize(frame, (model_size[0], model_size[1]))
+            img_str = cv2.imencode('.jpg', img)[1].tobytes()
+
+            response = inference(img_str, args['model_name'], grpc_stub)
+            postProcessMaskRCNN(response[0], response[1])
+        except Exception:
+            pass
```
Comment (brian-intel) [inline configs/opencv-ovms/scripts/grpc_python.py:85]:
```suggestion
            pass # nosec
```

**Review by jim-wang-intel — APPROVED** [review]
LGTM

**Pull request merged**